aten/src/ATen/native/Cross.cpp

```diff
@@ -1,52 +1,59 @@
 #include <ATen/ATen.h>
 #include <ATen/Dispatch.h>
 #include <ATen/NativeFunctions.h>
+#include <ATen/native/Resize.h>
+#include <ATen/ExpandUtils.h>
 
 #include <ATen/native/Cross.h>
 
 namespace at { namespace native {
 
 DEFINE_DISPATCH(cross_stub);
 
+int64_t _default_cross_dim(const c10::optional<int64_t> &dimension, IntArrayRef sizes) {
+  if(dimension.has_value()) {
+    return *dimension;
+  }
+
+  for(auto i : c10::irange(sizes.size())) {
+    if(sizes[i] == 3) {
+      return i;
+    }
+  }
+  TORCH_CHECK(false, "no dimension of size 3 in input");
+}
+
 Tensor cross(const Tensor & input, const Tensor & other, const c10::optional<int64_t> dimension) {
-  Tensor out = at::empty_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  native::cross_out(input, other, dimension, out);
-  return out;
+  auto dim = _default_cross_dim(dimension, input.sizes());
```

Comment:
Neat

Reply:
So this is a really nice refactor but I think it will break mobile (mobile is working on updating their architecture to handle refactors like this automatically).

cc @raziel @iseeyuan -- how should this PR be updated to handle this change? Should we revert it and continue relying on the old implementation for now with a TODO?

Reply:
To my understanding original `cross` is replaced by `linalg_cross`. It may break the existing models with `cross`. Are we keeping `aten::cross` while adding `linalg_cross`?

Reply:
We are keeping torch.cross (aten::cross) and adding torch.linalg.cross (aten::linalg_cross) but the internal implementation of aten::cross is being refactored to rely on linalg.cross